resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -157,7 +157,13 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private def allocateNewExecutorPod(): (String, Pod) = {
     val executorKubernetesId = UUID.randomUUID().toString.replaceAll("-", "")
```

Comment:
It's used as the identifier for each executor that the `KubernetesClusterSchedulerBackend` keeps in its `runningExecutorPods` map (identifier -> Pod).

Then when removing executors in `doKillExecutors` this is the identifier `CoarseGrainedSchedulerBackend` uses to identify which executors to be killed.

However, I don't see where the executorId we receive in `doKillExecutors` is ever given to the rest of Spark or the `CoarseGrainedSchedulerBackend`.  @mccheah is this a remnant of earlier versions where we aimed to support dynamic allocation?